channelserver/src/channelid.rs

```diff
@@ -0,0 +1,92 @@
+use std::fmt;
+
+use base64;
+use serde::ser::{Serialize, Serializer};
+use uuid::Uuid;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct ChannelID {
+    value: Uuid,
+}
+
+impl ChannelID {
+    pub fn is_valid(self) -> bool {
+        &self.value != &Uuid::nil()
+    }
+
+    pub fn to_string(self) -> String {
+        base64::encode_config(&self.value.as_bytes(), base64::URL_SAFE_NO_PAD)
+    }
+}
+
+impl Default for ChannelID {
+    fn default() -> Self {
+        Self {
+            value: Uuid::new_v4(),
```

Comment:
Are uuids now purely an implementation detail of this abstract `ChannelID` class?  If so, I wonder whether we need them at all, and if we could simplify this to just "generate 32 random bytes" rather than "generate a new v4 uuid".

Reply:
6 of 1, I suppose? I was mostly just stealing some of the UUID functions for things like `to_string()` and `nil()` but I can replicate those.